misc/testerina/modules/testerina-core/src/main/ballerina/types.bal

```diff
@@ -15,15 +15,19 @@
 // under the License.
 
 # Possible return types of the data provider function.
-public type DataProviderReturnType error|map<AnyOrError[]>|AnyOrError[][];
+public type DataProviderReturnType error|map<AnyOrErrorOrReadOnlyType[]>|AnyOrErrorOrReadOnlyType[][];
 
-public type AnyOrError any|error;
+public type AnyOrErrorOrReadOnlyType any|error|ReadOnlyType;
+
+public type ReadOnlyType readonly;
 
 # Represents errors generated by the Testerina module.
 public type TestError distinct error;
 
 type ExecutionError distinct error<record {|string functionName;|}>;
 
+public type serialExecutionType true|();
```

Comment:
Since we allowed `true`, users may be think `false` is also allowed.if we define like this, users will be confuse since there is no `false` allowed in here.

@MaryamZi @Dilhasha  wdyt?

Reply:
I think we said we'll change the field to `true serialExecution?;`, right? That will get rid of ambiguity.